app/code/community/Sendsmaily/Sync/Block/Rss.php

```diff
@@ -0,0 +1,131 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class Sendsmaily_Sync_Block_Rss extends Mage_Rss_Block_Catalog_Abstract
+{
+    const CACHE_TAG = 'block_html_sync_rss_index';
+
+    protected function _construct()
+    {
+        // Setting cache to save the RSS for 5 minutes.
+        $this->setCacheTags(array(self::CACHE_TAG));
+        $this->setCacheKey('sendsmaily_rss_index');
+        $this->setCacheLifetime(300);
+    }
+
+    protected function _toHtml()
+    {
+        $storeId = $this->_getStoreId();
+
+        // Header.
+        $title = Mage::app()->getStore()->getGroup()->getName();
+        $lang = Mage::getStoreConfig('general/locale/code');
+        $lastBuildDate = Mage::getSingleton('core/date')->date('D, d M Y H:i:s');
+
+        // Build array of elements.
+        $rssObj = Mage::getModel('sync/rss');
+        $data = array(
+            'title'         => $title,
+            'link'          => Mage::getBaseUrl(),
+            'description'   => 'Product Feed',
+            'lastBuildDate' => $lastBuildDate
+        );
+        $rssObj->_addHeader($data);
+
+        // Get store currency symbol.
+        $currencySymbol = Mage::app()
+            ->getLocale()
+            ->currency(
+                Mage::app()
+                ->getStore()
+                ->getCurrentCurrencyCode()
+            )->getSymbol();
+
+
+        // Gather products.
+        $product = Mage::getModel('catalog/product');
+
+        $products = $product->getCollection()
+            ->setStoreId($storeId)
+            ->addStoreFilter()
+            ->addAttributeToSelect(array('name', 'price', 'special_price', 'description', 'thumbnail'), 'inner')
+            ->addAttributeToSort('updated_at', 'desc');
+
+        $products->setVisibility(Mage::getSingleton('catalog/product_visibility')->getVisibleInSiteIds());
+        // Using resource iterator to load the data one by one instead of loading all at the same time.
+        Mage::getSingleton('core/resource_iterator')->walk(
+            $products->getSelect()->limit(50),
+            array(array($this, 'addNewItemXmlCallback')),
+            array('rssObj'=> $rssObj, 'product'=>$product, 'symbol' => $currencySymbol)
+        );
+
+        return $rssObj->createRssXml();
+    }
+
+    /**
+     * Preparing data and adding to RSS object
+     *
+     * @param array $args
+     */
+    public function addNewItemXmlCallback($args)
+    {
+        $product = $args['product'];
+        $currencySymbol = $args['symbol'];
+        Mage::dispatchEvent('sync_rss_new_xml_callback', $args);
+
+        $product->setData($args['row']);
+        // Show only active products.
+        if ($product->getStatus() === Mage_Catalog_Model_Product_Status::STATUS_ENABLED) {
+            $product->setAllowedInRss(true);
```

Comment:
Why not just return here? Is there a reason why you are setting `allowed_in_rss` to TRUE on the product.

Reply:
I was following Magento 1 original implementation of [RSS-feed array generation ](https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Rss/Block/Catalog/New.php).
```
    public function addNewItemXmlCallback($args)
    {
        $product = $args['product'];

        $product->setAllowedInRss(true);
        $product->setAllowedPriceInRss(true);
        Mage::dispatchEvent('rss_catalog_new_xml_callback', $args);

        if (!$product->getAllowedInRss()) {
            //Skip adding product to RSS
            return;
        }
        $allowedPriceInRss = $product->getAllowedPriceInRss();
```
I think they use it to change behaviour of RSS-product displaying by manipulating the dispatched event.

As I found out that the default collection gives out only enabled products, so there is actually no need for the enabled status check there. I think I should strip out both.
https://github.com/sendsmaily/Sendsmaily-Sync-for-Magento/blob/b893bfab20bfcb48fb046b8b142448582b316257/app/code/community/Sendsmaily/Sync/Block/Rss.php#L95-L104

Reply:
OK. If it doesn't change the output of the RSS feed, then it would best if the logic is removed entirely (for clarity).

Reply:
Removed both in https://github.com/sendsmaily/Sendsmaily-Sync-for-Magento/pull/6/commits/3261ce594e50590e625f83bb26fedc30a21bd48f